src/algebra/star/algebra.lean

```diff
@@ -15,31 +16,56 @@ Introduces the notion of a star algebra over a star ring.
 
 universes u v w
 
+open opposite
+
 /--
 A star algebra `A` over a star ring `R` is an algebra which is a star ring,
 and the two star structures are compatible in the sense
 `star (r • a) = star r • star a`.
 -/
--- Note that we take `star_ring A` as a typeclass argument, rather than extending it,
+-- Note that we take `star_monoid A` as a typeclass argument, rather than extending it,
 -- to avoid having multiple definitions of the star operation.
-class star_algebra (R : Type u) (A : Type v)
-  [comm_semiring R] [star_ring R] [semiring A] [star_ring A] [algebra R A] :=
+class star_algebra (R : Type u) (A : Type v) [has_star R] [has_star A] [has_scalar R A] :=
 (star_smul : ∀ (r : R) (a : A), star (r • a) = (@has_star.star R _ r) • star a)
 
 @[simp] lemma star_smul {R : Type u} {A : Type v}
-  [comm_semiring R] [star_ring R] [semiring A] [star_ring A] [algebra R A] [star_algebra R A]
+  [has_star R] [has_star A] [has_scalar R A] [star_algebra R A]
   (r : R) (a : A) :
   star (r • a) = star r • star a :=
 star_algebra.star_smul r a
 
+attribute [instance] ring_hom_inv_pair.of_ring_equiv
+
+/-- If `A` is a left- and right- module over a commutative `R` with compatible actions,
+then `star` is a semilinear equivalence. -/
+@[simps]
+def star_linear_equiv {R : Type u} {A : Type v}
+  [comm_semiring R] [star_ring R] [semiring A] [star_ring A]
+  [module R A] [star_algebra R A] :
+    A ≃ₛₗ[((star_ring_aut : ring_aut R) : R →+* R)] A :=
```

Comment:
In the case of the trivial star operation (like we'd choose for the reals), we want this to be exactly the same as `refl :  A ≃ₗ[R] A`.  I'm worried that that's not the case currently:  doesn't it turn out that the coercion `(ring_equiv.refl R : R →+* R)` is not defeq to `ring_hom.id R`?

If my concern is correct, this can be fixed by defining
```lean
def star_ring_end [comm_semiring R] [star_ring R] : R →+* R :=
{ to_fun := star,
  ..star_add_equiv,
  ..star_mul_aut }
```
which actually is defeq to `ring_hom.id R` in the case of trivial-star, and 
```lean
instance [comm_semiring R] [star_ring R] :
  ring_hom_inv_pair (star_ring_end R) (star_ring_end R) :=
⟨ring_hom.ext star_star, ring_hom.ext star_star⟩
```